src/vellum/workflows/nodes/displayable/bases/prompt_deployment_node.py

```diff
@@ -76,6 +87,49 @@ def _get_prompt_event_stream(self) -> Iterator[ExecutePromptEvent]:
             request_options=request_options,
         )
 
+    def _process_prompt_event_stream(self) -> Generator[BaseOutput, None, Optional[List[PromptOutput]]]:
```

Comment:
[nit] is there a way for us to reuse `super(). _process_prompt_event_stream()`? lots of reused logic seemingly. no worries if too complex